src/components/auth/Register.jsx

```diff
@@ -1,9 +1,332 @@
+            "firstName": firstName.trim(),
+            "lastName": lastName.trim(),
+            "email": email.trim(),
+            "password": password,
+            "dateOfBirth": dateOfBirth,
+            "adultName": adultName.trim(),
+          };
+          result = await registerStudent(studentData);
+        } else {
+          let teacherData = {
+            "firstName": firstName,
+            "lastName": lastName,
+            "email": email,
+            "password": password,
+          };
+          result = await registerTeacher(teacherData);
+        }
+        console.log(result);
+        if (result.status === 201) {
+          setIsModalOpen(true);
+        }
+      } catch (error) {
+        console.log('ERROR');
+        console.log(error);
+        if (error.message.includes("taken")) {
+          setEmailError(error.message);
+        }
+      }
+    }
+
+
+  }
+
+  const switchUserRole = () => {
+    if (userRole === 'student') {
+      setUserRole('teacher');
+    } else {
+      setUserRole('student');
+    }
+  }
+
+  const handleFirstNameChange = (event) => {
+    const newFirstName = event.target.value;
+    setFirstName(newFirstName);
+  }
+
+  const handleLastNameChange = (event) => {
+    const newLastName = event.target.value;
+    setLastName(newLastName);
+  }
+
+  const handleDateOfBirthChange = (event) => {
+    const newBirthDate = event.target.value;
+    setDateOfBirth(newBirthDate);
+    const age = calculateAge(newBirthDate);
+    setIsAdultNameRequired(age < 16);
+    setAdultNameError('');
+  };
+
+  const handleAdultNameChange = (event) => {
+    setAdultName(event.target.value);
+  };
+
+  const handleEmailChange = (event) => {
+    const newEmail = event.target.value;
+    setEmail(newEmail);
+    if (newEmail.trim() !== '') {
+      setEmailError('');
+    }
+  }
+
+  const handleConfirmEmailChange = (event) => {
+    setConfirmEmail(event.target.value);
+  }
+
+  const handlePasswordChange = (event) => {
+    const newPassword = event.target.value;
+    setPassword(newPassword);
+    if (newPassword.trim() !== '') {
+      setPasswordError('');
+    }
+  }
+
+  const handleConfirmPasswordChange = (event) => {
+    setConfirmPassword(event.target.value);
+  }
+
+  const handleGotoLogin = () => {
+    setIsModalOpen(false);
+    navigate('/login');
+  }
+
+  const calculateAge = (birthdate) => {
+    const birthDate = new Date(birthdate);
+    const currentDate = new Date();
+
+    let age = currentDate.getFullYear() - birthDate.getFullYear();
+
+    const currentMonth = currentDate.getMonth();
+    const currentDay = currentDate.getDate();
+    const month = birthDate.getMonth();
+    const day = birthDate.getDay();
+
+    if (currentMonth < month - 1 || (currentMonth === month - 1 && currentDay < day)) {
+      age--;
+    }
+
+    return age;
+  }
+
   return (
-    <>
-      <h1>Registration Form</h1>
-    </>
+    <section className="w-full max-w-xl mx-auto px-10 py-5">
```

Comment:
You can move the return to the separate file RegisterForm.jsx and leave only the functional part in Register.jsx

Reply:
Done. 
Separate component RegisterForm was created